tasks/task_16/provider/google_provider/google_provider.go

```diff
@@ -0,0 +1,37 @@
+package google_provider
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/drive/v2"
+	"studing/tasks/task_16/configs"
+)
+
+type GoogleProvider struct {
+	client *http.Client
+}
+
+func NewGoogleProvider(ctx context.Context, c *configs.Config) (*GoogleProvider, error) {
```

Comment:
why configs.Config is a pointer?

Reply:
configs.Config is declared in another package, so here you have dependency between packages, which can be avoided (once config is fully used by the provider - it can be declared here, another approach - to pass scalars)